### PR TITLE
Mqtt expansion

### LINKED ISSUE #50: Optimize and add functionality to MQTTController
## Description

MQTTController needs to be updated to prevent connection issues, apparent solution at the time of writing is the utilization of the singleton pattern, Additionally, due to this new implementation, additional constraints will need to be added for text views. To prevent duplicate entries across multiple activity initializations when the application changes screens.

## Checklist

- [x]  The singleton pattern works perfectly with all classes
- [x]  No duplicate entries across multiple activity initializations.

[Part of Epic Feature 5](https://github.com/DIT113-V22/group-04/wiki/EF-5-Connectivity)

### PR DESCRIPTION
**Pull request by Soarnir**
# Description

Converts the mqtt controller to an object and implements the singleton pattern to manage the same connection across multiple screens, as a bonus, this also solved some connection issues apparent in the current implementation.
Additionally, an update to gradle has been included.

## Issues

- Closes #50 

## Dependencies

- Updated Gradle from 7.1.3 to 7.2.0

# Checklist

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Documentation updated to cover the changes implemented (including changes to the wiki)
- [x] Code change tested manually if not covered by automated tests
- [x] Specified any additions, alterations or removals of dependencies
- [x] Followed the code style present in the project
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: android/Drawer/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.example.drawer">
```
Comment (Kev049) [inline android/Drawer/app/src/main/AndroidManifest.xml:3]: @Soarnir Could you please elaborate on why this line was removed/moved to the build.gradle?

Comment (Soarnir) [reply android/Drawer/app/src/main/AndroidManifest.xml:3]: New standard from gradle and android states that the manifest should be declared in the build.gradle instead, so it's just a case of following industry standards. We couldn't do this prior to 7.2.0, so I implemented it now : )

Comment (Soarnir) [reply android/Drawer/app/src/main/AndroidManifest.xml:3]: And naturally, it's a case of keeping things up to date, especially regarding our build tools.

Comment (Kev049) [reply android/Drawer/app/src/main/AndroidManifest.xml:3]: Alright, that makes perfect sense! Thank you for the clarification!

Comment (Kev049) [reply android/Drawer/app/src/main/AndroidManifest.xml:3]: Everything else looks good to me, so am approving the pull request. :)

**Review by Kev049 — APPROVED** [review]

**Pull request merged**